tests/run/duplicate_utilitycode.pyx

```diff
@@ -0,0 +1,27 @@
+# mode: run
+
+# There were a few cases where duplicate utility code definitions (i.e. with the same name)
+# could be generated, causing C compile errors. This file tests them.
+
+from __future__ import print_function
+
+cdef f1(x, r):
+    return "f1"
+
+cdef f2(x1, r):
```

Comment:
In this case the reason that the two bits of utilitycode don't match (and so it's generated "twice") is that the arguments have different names. One generates:

```
def wrap(object x, object r)
```

and the other

```
def wrap(object x1, object r)
```

There's a few options here:
1. Don't include the argument names in the wrapping: they're pure C functions and so the arguments should be viewed as positional only. The utility code will be detected as identical and eliminated. This is my view, but it may break existing code.
2. Include the argument names in the generated c name (since it's part of the "signature" kind of) thus creating two different conversion functions but making them unique.
3. Add a number to the end of the function if needed (what this PR is currently doing).